### ISSUE TITLE
Failure with '_' instead of variable.

### ISSUE THREAD

**Issue by JalonSolov** [vlang/v#4993]
**V version:** V 0.1.27 88fa935
**OS:** Ubuntu 20.04

**What did you do?**

Tiny program I was playing around with, to see if variables could be used for `len:` and `init:` when declaring arrays:
```
fn main() {
        l := 20
        i := `a`
        for _ in 0..10000 {
                s := []byte{len:l, init:i}
        }
}
```
With the `s` in place, v tells me "unused variable: \`s\`" which is correct.  However if I then replace the `s` with `_`, bad things happen.

**What did you expect to see?**

Nothing.  Just return to prompt.
 
**What did you see instead?**
```
jalon@Ripper:~/git/v$ v bar.v
==================
/home/jalon/.cache/v/bar.tmp.c: In function ‘main’:
/home/jalon/.cache/v/bar.tmp.c:9728:64: error: ‘_val_59’ undeclared (first use in this function)
 9728 |  {array_byte _ = __new_array_with_default(l, 0, sizeof(byte), &_val_59);}
      |                                                                ^~~~~~~
/home/jalon/.cache/v/bar.tmp.c:9728:64: note: each undeclared identifier is reported only once for each function it appears in
/home/jalon/.cache/v/bar.tmp.c: In function ‘vcalloc’:
/home/jalon/.cache/v/bar.tmp.c:4720:1: warning: control reaches end of non-void function [-Wreturn-type]
 4720 | }
      | ^
/home/jalon/.cache/v/bar.tmp.c: In function ‘byte_is_white’:
/home/jalon/.cache/v/bar.tmp.c:7491:1: warning: control reaches end of non-void function [-Wreturn-type]
 7491 | }
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

jalon@Ripper:~/git/v$
```

**Comment by spytheman** [comment]
An minimal testcase is: `_ := []byte{init:1}`, which produces:
```c
0[17:28:48]delian@nemesis: /v/vv $ ./v -printfn main -o z.c run z.v
int main(int ___argc, char** ___argv){
        _vinit();
{array_byte _ = __new_array_with_default(0, 0, sizeof(byte), &_val_7);}
;
        return 0;
}
```

**Comment by spytheman** [comment]
Fixed in: aef751861db0b6a420d95036f13c75bd506f3d7c
